### PR TITLE
OCRVS-3771 - `<Toast>` & `<Alert>`

### LINKED ISSUE #3771: Components / Toast
We have three Toast components:

- client/src/components/interface/ToastNotification.tsx
- client/src/components/Notification.tsx
- /components/src/components/interface/ErrorToastNotification.tsx
- components/src/components/interface/FloatingNotification.tsx

This ticket is too refactor into one component. Toast
https://www.figma.com/file/SsqCzO2eherXXCKyixmli9/OpenCRVS-Design-System-v2-WIP?node-id=37%3A7763

- updates styling
- maxWidth on desktop 520px
- Fixes centring on page. aligned to Header. Not screen width
- can take a type prop (Positive, Neutral, Warning, Loading)
- option to have Retry button
- Update instances to use Toast
- Add auto hide after 3 seconds...?

### Storybook
Data / Toast
- Example of each

Tasks
[ ]

### LINKED ISSUE #3773: Components / Alert
We currently have a component called Warning. Which acts as an Alert. 

This ticket is to create/refactor Warning into an Alert component:
https://www.figma.com/file/SsqCzO2eherXXCKyixmli9/OpenCRVS-Design-System-v2-WIP?node-id=25%3A5964
- Update styling. Similar to Toast.
- Adds additional variants (Positive, Neutral, Warning, Default)
- Width is based off the container. eg Side Nav or Dialog width. 
- Close button is optional. Default off.

### Storybook
Data / Alert
- Examples of each variant

### PR DESCRIPTION
**Pull request by naftis**
Closes #3771
Closes #3773

* Improves `<Toast>` documentation
* Refactored the component look
* Added action button which usually is used for Retrying
* Refactors `<Toast>` to use `<Alert>` with fixed & animated styles
* Renames client's `ToastNotification` to `GenericErrorToast`

### CODE REVIEW

File: packages/client/src/views/SysAdmin/Config/Forms/Home/ActionsNotification.tsx [hunk 1]
```diff
@@ -35,24 +36,25 @@ export function ActionsNotification() {
...
-        intl.formatMessage(statusChangeActionMessages(action)[status], {
-          event: intl.formatMessage(constantsMessages[event]),
-          version
-        })}
-    </Toast>
+    <>
+      {isNotifiable(status) && (
+        <Toast
+          type={
+            isNotifiable(status) ? NOTIFICATION_TYPE_MAP[status] : 'warning'
```
Comment (Zangetsu101) [inline packages/client/src/views/SysAdmin/Config/Forms/Home/ActionsNotification.tsx:43]: I think we can safely remove the `isNotifiable(status)` check from here as we are doing so before calling the Toast. Nice job removing the `show` prop from the `Toast` component. It made some of the checks like this not needed anymore.

File: packages/client/src/views/SysAdmin/Config/Forms/Home/ActionsNotification.tsx [hunk 2]
```diff
@@ -35,24 +36,25 @@ export function ActionsNotification() {
...
+          }
+          onClose={
+            status !== ActionStatus.PROCESSING
+              ? () => setAction({ status: ActionStatus.IDLE })
+              : noop // Toast isn't closable when status is "loading"
+          }
+        >
+          {isNotifiable(status) &&
```
Comment (Zangetsu101) [inline packages/client/src/views/SysAdmin/Config/Forms/Home/ActionsNotification.tsx:51]: Same as above

File: packages/client/src/views/SysAdmin/Config/Forms/Wizard/SaveActionNotification.tsx [hunk 3]
```diff
@@ -16,29 +16,32 @@ import {
...
-          : undefined
-      }
-    >
-      {isNotifiable(status) && intl.formatMessage(saveActionMessages[status])}
-    </Toast>
+    <>
+      {isNotifiable(status) && (
+        <Toast
+          type={
+            isNotifiable(status) ? NOTIFICATION_TYPE_MAP[status] : 'warning'
```
Comment (Zangetsu101) [inline packages/client/src/views/SysAdmin/Config/Forms/Wizard/SaveActionNotification.tsx:33]: This one as well

File: packages/components/src/Toast/Toast.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+<Source
+  dark
+  code={`
+{isVisible && (
+  <Toast
+    type="success"
+    onActionClick={() => alert('Retried')}
+    actionText="Retry"
+    onClose={() => setVisible(false)}
```
Comment (Zangetsu101) [inline packages/components/src/Toast/Toast.stories.mdx:126]: I think we don't want the dismiss action here?

Comment (naftis) [reply packages/components/src/Toast/Toast.stories.mdx:126]: @jpye-finch are the retriable toasts dismissable? My thought process was that everything would be dismissable except loading ones

**Pull request merged**